src/crypto-providers/trezorCryptoProvider.ts

```diff
@@ -278,30 +341,33 @@ const TrezorCryptoProvider: () => Promise<CryptoProvider> = async () => {
       stakeSigningFiles,
     } = filterSigningFiles(signingFiles)
     const inputs = txAux.inputs.map(
-      (input, i) => prepareInput(input, getSigningPath(paymentSigningFiles, i)),
+      (input: _Input, i: number) => prepareInput(input, getSigningPath(paymentSigningFiles, i)),
     )
     const outputs = txAux.outputs.map(
-      (output) => prepareOutput(output, network, changeOutputFiles),
+      (output: _Output) => prepareOutput(output, network, changeOutputFiles),
     )
     const certificates = txAux.certificates.map(
-      (certificate) => prepareCertificate(certificate, stakeSigningFiles),
+      (certificate: _Certificate) => prepareCertificate(certificate, stakeSigningFiles),
     )
-    const fee = `${txAux.fee}`
-    const ttl = `${txAux.ttl}`
+    const fee = txAux.fee.toString()
+    const ttl = prepareTtl(txAux.ttl)
+    const validityIntervalStart = prepareValidityIntervalStart(txAux.validityIntervalStart)
     const withdrawals = txAux.withdrawals.map(
-      (withdrawal) => prepareWithdrawal(withdrawal, stakeSigningFiles),
+      (withdrawal: _Withdrawal) => prepareWithdrawal(withdrawal, stakeSigningFiles),
     )
 
-    const response = await TrezorConnect.cardanoSignTransaction({
+    const response = await TrezorConnect.cardanoSignTransaction(removeNullFields({
```

Comment:
why do you need to remove null fields? was there any issue with them? If so, that can be a validation bug on trezor-connects side which I can fix

Reply:
I think that might be the case, when I don't remove the null fields I get
```
Error: Parameter "ttl" has invalid type. "string" expected.
```

Reply:
I experienced this as well, for some reason, trezor don't like even the optional parameters to be undefined and requires them to be missing.

Reply:
Indeed, you are right, Trezor's common paramsValidator enforces the specific type even if the field is not obligatory :/ https://github.com/trezor/connect/blob/develop/src/js/core/methods/helpers/paramsValidator.js#L22

Reply:
I created an issue in connect's repo to get the insight of their developers whether it's worth addressing systematically: https://github.com/trezor/connect/issues/770